docs/architecture/psa-migration/dependencies-tls.md

```diff
@@ -0,0 +1,563 @@
+mbedtls_cipher_free
+mbedtls_cipher_info_from_type
+mbedtls_cipher_init
+mbedtls_cipher_set_padding_mode
+mbedtls_cipher_setkey
+mbedtls_cipher_setup
+
+mbedtls_dhm_calc_secret
+mbedtls_dhm_free
+mbedtls_dhm_get_bitlen
+mbedtls_dhm_get_len
+mbedtls_dhm_get_value
+mbedtls_dhm_init
+mbedtls_dhm_make_params
+mbedtls_dhm_make_public
+mbedtls_dhm_read_params
+mbedtls_dhm_read_public
+mbedtls_dhm_set_group
+
+mbedtls_ecdh_calc_secret
+mbedtls_ecdh_free
+mbedtls_ecdh_get_params
+mbedtls_ecdh_init
+mbedtls_ecdh_make_params
+mbedtls_ecdh_make_public
+mbedtls_ecdh_read_params
+mbedtls_ecdh_read_public
+mbedtls_ecdh_setup
+
+mbedtls_ecjpake_check
+mbedtls_ecjpake_derive_secret
+mbedtls_ecjpake_free
+mbedtls_ecjpake_init
+mbedtls_ecjpake_read_round_one
+mbedtls_ecjpake_read_round_two
+mbedtls_ecjpake_set_point_format
+mbedtls_ecjpake_setup
+mbedtls_ecjpake_write_round_one
+mbedtls_ecjpake_write_round_two
+
+mbedtls_ecp_curve_info_from_grp_id
+mbedtls_ecp_curve_info_from_tls_id
+
+mbedtls_md_clone
+mbedtls_md_finish
+mbedtls_md_free
+mbedtls_md_get_size
+mbedtls_md_get_type
+mbedtls_md_hmac_finish
+mbedtls_md_hmac_reset
+mbedtls_md_hmac_starts
+mbedtls_md_hmac_update
+mbedtls_md_info_from_type
+mbedtls_md_init
+mbedtls_md_setup
+mbedtls_md_starts
+mbedtls_md_update
+
+mbedtls_mpi_bitlen
+mbedtls_mpi_free
+mbedtls_mpi_read_binary
+
+mbedtls_pk_can_do
+mbedtls_pk_debug
+mbedtls_pk_decrypt
+mbedtls_pk_encrypt
+mbedtls_pk_get_bitlen
+mbedtls_pk_sign
+mbedtls_pk_sign_restartable
+mbedtls_pk_verify
+mbedtls_pk_verify_restartable
+
+mbedtls_sha256_clone
+mbedtls_sha256_finish
+mbedtls_sha256_free
+mbedtls_sha256_init
+mbedtls_sha256_starts
+mbedtls_sha256_update
+
+mbedtls_sha512_clone
+mbedtls_sha512_finish
+mbedtls_sha512_free
+mbedtls_sha512_init
+mbedtls_sha512_starts
+mbedtls_sha512_update
+```
+
+Note: the direct dependency on MPI functions is in order to manage DHM
+parameters, that are currently stored as a pair of MPIs in the
+`mbedtls_ssl_config` structure. (The public API uses byte arrays or a
+`mbedtls_dhm_context` structure.)
+
+Note: the direct dependency on ECP APIs is in order to access information;
+no crypto operation is done directly via this API, only via the PK and ECDH
+APIs.
+
+Note: the direct dependencies on the SHA-2 modules instead of using the
+MD layer is for convenience (and perhaps to save some memory as well) and can
+easily be replace by use of a more generic API.
+
+Key exchanges and other configuration options
+---------------------------------------------
+
+In the file-level analysis below, many things are only used if certain key
+exchanges or other configuration options are enabled. This section sums up
+those key exchanges and options.
+
+Key exchanges:
+
+- DHE-PSK
+- DHE-RSA
+- ECDH-ECDSA
+- ECDH-RSA
+- ECDHE-ECDSA
+- ECDHE-PSK
+- ECDHE-RSA
+- ECJPAKE
+- PSK
+- RSA
+- RSA-PSK
+
+Protocol:
+
+- `MBEDTLS_SSL_PROTO_TLS1_2`
+- `MBEDTLS_SSL_PROTO_DTLS`
+- `MBEDTLS_SSL_PROTO_TLS1_3_EXPERIMENTAL` (excluded from this analysis)
```

Comment:
As Gilles points out, this document is prone to becoming out-of-date very quicly, and since it was only mostly there just to support my investigation until a list of tasks and potential problems could be established, which is now done, we can probably just remove the whole document, unless you have an objection. (Same for dependencies-x509 obviously.)

Reply:
Yes, let's remove them, rather than keep it something that quickly becomes obsolete. Their content has now been distilled into strategy.md + github issues.

Reply:
Agreed